lib/pluginmanager/sources/http.rb

```diff
@@ -0,0 +1,78 @@
+# encoding: utf-8
+require "rubygems/package"
+require "pluginmanager/sources/base"
+require "net/http"
+
+module LogStash::PluginManager::Sources
+
+  class HTTP < Base
+
+    attr_reader :uri, :fallback, :header_options
+
+    ##
+    # Fallback URL used by default when installing packs given
+    # by name, if you require to install a pack from a custom URL
+    # you will need to pass the full URL.
+    ##
+    ROOT_URI = "https://download.elastic.co".freeze
+
+    ##
+    # User agent string used to identify the requester is a
+    # LogStash agent
+    ##
+    USER_AGENT = "Logstash/#{LOGSTASH_VERSION}".freeze
+
+    def initialize(uri)
+      super
+      @fallback = !uri.start_with?("http")
+      if fallback
+        @uri = URI(fallback_uri(uri))
+      else
+        @uri = URI(uri)
+      end
+      @header_options = {'User-Agent' => USER_AGENT}
+    end
```

Comment:
After looking at the implementation of https://github.com/elastic/logstash/pull/4584/files#diff-a216a3887000aed590754e380e388739R96

I don't think the `fallback` will be called.